pkg/csi/service/vanilla/controller.go

```diff
@@ -926,39 +926,112 @@ func (c *controller) createFileVolume(ctx context.Context, req *csi.CreateVolume
 			"parsing storage class parameters failed with error: %+v", err)
 	}
 
-	var createVolumeSpec = common.CreateVolumeSpec{
-		CapacityMB: volSizeMB,
-		Name:       req.Name,
-		ScParams:   scParams,
-		VolumeType: common.FileVolumeType,
+	// Check if vCenter task for this volume is already registered as part of
+	// improved idempotency CR
+	log.Debugf("Checking if vCenter task for file volume %s is already registered.", req.Name)
+	var (
+		volTaskAlreadyRegistered bool
+		faultType                string
+		volumeID                 string
+		vcenter                  *cnsvsphere.VirtualCenter
+	)
+	// Get VirtualCenter instance
+	vcenter, err = c.manager.VcenterManager.GetVirtualCenter(ctx, c.manager.VcenterConfig.Host)
+	if err != nil {
+		return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
+			"failed to get vCenter. Err: %v", err)
+	}
+	// Get operation store
+	operationStore := c.manager.VolumeManager.GetOperationStore()
+	if operationStore == nil {
+		return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
+			"Operation store cannot be nil")
 	}
-	var volumeID string
-	var faultType string
-	filterSuspendedDatastores := commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.CnsMgrSuspendCreateVolume)
-	if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.CSIAuthCheck) {
-		fsEnabledClusterToDsInfoMap := c.authMgr.GetFsEnabledClusterToDsMap(ctx)
 
-		var filteredDatastores []*cnsvsphere.DatastoreInfo
-		for _, datastores := range fsEnabledClusterToDsInfoMap {
-			filteredDatastores = append(filteredDatastores, datastores...)
-		}
+	volumeOperationDetails, err := operationStore.GetRequestDetails(ctx, req.Name)
+	if err != nil {
```

Comment:
Can we check if this err is for Not Found error from API server and not any other error?
Only in that case you can print error and continue create a new task.
Can we make the same fix for block vanilla as well?

Reply:
done. I am handling Not found error and any other error separately now.